src/plugins/template/backend/int_executable.cpp

```diff
@@ -83,14 +83,20 @@ class TemporaryOverrideOutputs {
         for (size_t i = 0; i < args.size(); ++i) {
             auto output = node->get_input_source_output(i);
```

Comment:
@ilyachur For parameter node, error happened at here, `i = 0`, get `**node input index is out of range**` ERROR. The `Parameter Node` no `get_input_source_output(0)` right?

Reply:
@ilyachur  & @jane-intel  Try the following test, find only change the `Parameter node` shape can not infer the other nodes shape.
```
    std::vector<TemporaryOverrideOutputs> overrider_vec = {};
    for (const auto& param : get_parameters()) {
        // auto output = node->get_input_source_output(i);
        std::cout << "param: " << param->get_name()
                  << " default shape: " << param->get_output_partial_shape(0)
                  << " new shape: " << tensor_map.at(param->output(0).get_tensor_ptr()).get_shape() << std::endl;
        param->set_partial_shape(tensor_map.at(param->output(0).get_tensor_ptr()).get_shape());
        // parameter->set_partial_shape(orig_shapes[i]);
        param->validate_and_infer_types();
        // overrider_vec.push_back(TemporaryOverrideOutputs(param, {tensor_map.at(param->output(0).get_tensor_ptr())}));
        for (const auto& op : m_nodes) {
            std::cout << "[TEST OP] " << op->get_name() << std::endl;
            for (auto input : op->inputs()) {
                if (tensor_map.find(input.get_tensor_ptr()) == tensor_map.end())
                    std::cout << " - inputs:     " << input.get_partial_shape() << std::endl;
                else
                    std::cout << " - inputs:     " << input.get_partial_shape()
                              << "\t tensor_map: " << tensor_map.at(input.get_tensor_ptr()).get_shape() << std::endl;
            }
        }
    }
```
**This is the original graph nodes shape output:**
![image](https://user-images.githubusercontent.com/52686861/228559945-0acf2e69-6604-4d0b-ac67-19a74cc27315.png)
**This is the graph nodes shape after set the first `Parameter node` shape:**
![image](https://user-images.githubusercontent.com/52686861/228560604-a885ae6a-6d30-46af-95e9-2a46af935f6c.png)
 We can find the `ReduceMean_1442` node change shape from dynamic to static success **([?,2,?,?] ->[1, 2, 3, 4])**, but the next node `Subtract_1443` failed, the first input shape change success **([?,2,?,?] ->[1, 2, 3, 4])** but the second input failed **([?,2,1,1] ->NO CHANGE)**. The second input shape needs the output shape from previous node `ReduceMean_1442` I think, but no such info. 

**The func `validate_and_infer_types()` can only infer the current node shape not all nodes in the graph, am I right?**

Reply:
Yes you need to call the model method

Reply:
@jane-intel I thought that shape propagation should work in this case... Am I wrong?

Reply:
@zhaixuejun1993 Please push the latest changes to this branch. It is needed in order to provide any advice.
Right now I see that situation looks like shapes weren't propagated from the parameters to other operations.

Also after `set_partial_shape()` for all parameters you MUST call `ov::Model::validate_and_infer_types()`.

Reply:
The root cause is the func `validate_nodes_and_infer_types()` call.
First set all parameter node shape, call `validate_nodes_and_infer_types()` by one of the parameter node failed, but by model success.